src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private T[] array;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increaseSize();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAddIndex(index);
+        increaseSize();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
```suggestion
        for (T element : list) {
            add(element);
        }
```
You can use for each loop here

Reply:
Did I understand correctly that to implement this code block 

for (T element : list) {
add(element);
}

 I need to implement the Iterable interface? Or would it be better to do it some other way?